src/main/java/com/github/dikhan/domain/Payload.java

```diff
@@ -0,0 +1,270 @@
+         * @return Payload Builder with timestamp field populated to be able to keep populating the instance
+         */
+        public Builder setSummary(String summary) {
+            this.summary = summary;
+            return this;
+        }
+
+        /**
+         * @param source The unique location of the affected system, preferably a hostname or FQDN.
+         * @return Payload Builder with timestamp field populated to be able to keep populating the instance
+         */
+        public Builder setSource(String source) {
+            this.source = source;
+            return this;
+        }
+
+        /**
+         * @param severity The perceived severity of the status the event is describing with respect to the affected
+         *                 system. This can be critical, error, warning or info.
+         * @return Payload Builder with timestamp field populated to be able to keep populating the instance
+         */
+        public Builder setSeverity(Severity severity) {
+            this.severity = severity;
+            return this;
+        }
+
+        /**
+         * @param timestamp The time at which the emitting tool detected or generated the event in ISO 8601 format.
+         * @return Payload Builder with timestamp field populated to be able to keep populating the instance
+         */
+        public Builder setTimestamp(OffsetDateTime timestamp) {
+            this.timestamp = timestamp;
+            return this;
+        }
+
+        /**
+         * @param component Component of the source machine that is responsible for the event, for example mysql or eth0.
+         * @return Payload Builder with component field populated to be able to keep populating the instance
+         */
+        public Builder setComponent(String component) {
+            this.component = component;
+            return this;
+        }
+
+        /**
+         * @param group Logical grouping of components of a service, for example app-stack.
+         * @return Payload Builder with group field populated to be able to keep populating the instance
+         */
+        public Builder setGroup(String group) {
+            this.group = group;
+            return this;
+        }
+
+        /**
+         * @param eventClass The class/type of the event, for example ping failure or cpu load.
+         * @return Payload Builder with eventClass field populated to be able to keep populating the instance
+         */
+        public Builder setEventClass(String eventClass) {
+            this.eventClass = eventClass;
+            return this;
+        }
+
+        /**
+         * @param customDetails An arbitrary JSON object containing any data you'd like included in the incident log.
+         * @return Payload Builder with customDetails field populated to be able to keep populating the instance
+         */
+        public Builder setCustomDetails(String customDetails) {
```

Comment:
We should probably enforce this input to be Json object to meet the field specification requirements:

Custom Details, Free-form details from the event,  Object,  Optional,  {"ping time": "1500ms", "load avg": 0.75 }

When building the Payload we can keep it still as string as the verification would have been done within the builder itself.

Thoughts?